spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/Spark3Util.java

```diff
@@ -937,6 +938,28 @@ public static org.apache.spark.sql.catalyst.TableIdentifier toV1TableIdentifier(
     return org.apache.spark.sql.catalyst.TableIdentifier.apply(table, database);
   }
 
+  public static String determineWriteBranch(
```

Comment:
This method can technically be embedded in `SparkWriteConf.branch()`, and we can also remove `SparkWriteConf.wapBranch()` method. The only issue is that it will compute `wapId` and `wapEnabled` twice, once in the method, once in `SparkWrite`/`SparkPositionDeltaWrite`. Please let me know if there is any preference for code cleanness vs computation.

Reply:
I think it makes more sense to embed this in `SparkWriteConf.branch()`. `wapId` and `wapEnabled` are constant so it should be fine.

Reply:
Yeah just took a look in the code, there's no real computation for `wapId` and `wapEnabled` they're just constants, so I'm in favor of having it in `SparkWriteConf` as well.